test/e2e/specs/fakerp/setcontainerimage.go

```diff
@@ -0,0 +1,62 @@
+package fakerp
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	pluginapi "github.com/openshift/openshift-azure/pkg/api/plugin"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/sanity"
+)
+
+var _ = Describe("Change a single image to latest E2E tests [ChangeImage][Fake][LongRunning]", func() {
+	var (
+		azurecli *azure.Client
+		ctx      = context.Background()
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment(false)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be possible for an SRE to update a single container image", func() {
+		By("Reading the internal config before the image update")
+		before, err := azurecli.OpenShiftManagedClustersAdmin.Get(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(before).NotTo(BeNil())
+
+		By("Executing a cluster update with updated image.")
+
+		data, err := ioutil.ReadFile("../../pluginconfig/pluginconfig-311.yaml")
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
`Expect(data).NotTo(BeNil())`

Reply:
thanks, i'll add these checks

Reply:
@charlesakalugwu really, no.  There is no justification for not trusting that the underlying library will return data if it does not return a nil error.

Reply:
or any of the code we write either.
If we get an unexpected NPE in tests, so be it.